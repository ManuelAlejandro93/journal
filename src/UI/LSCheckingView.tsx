import { CircularProgress, Typography } from '@mui/material';
import { AuthLayout } from '@/Auth';

export const LSCheckingView = () => {
  return (
    <AuthLayout authPageName={``}>
      <div className='flex flex-col items-center'>
        <Typography fontSize={30}>
          Checking your credentials, plase wait.
        </Typography>
        <CircularProgress size={50}></CircularProgress>
      </div>
    </AuthLayout>
  );
};
