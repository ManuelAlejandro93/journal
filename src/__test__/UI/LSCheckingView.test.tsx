import { render, screen } from '@testing-library/react';
import { LSCheckingView } from '../../UI/LSCheckingView';

describe('Pruebas en <LSCheckingView/>', () => {
  test('Debe ser renderizable', () => {
    screen;
    const { container } = render(<LSCheckingView />);
    expect(container).toBeTruthy();
  });

  test('Debe ser igual al snapshot', () => {
    const { container } = render(<LSCheckingView />);
    expect(container).toMatchSnapshot();
  });

  test('El texto "Checking your credentials, plase wait." debe ser visible en el documento', () => {
    const {} = render(<LSCheckingView />);
    const elementoConTexto = screen.getByText(
      /Checking your credentials, plase wait./i
    );

    expect(elementoConTexto).toBeVisible();
  });
});
