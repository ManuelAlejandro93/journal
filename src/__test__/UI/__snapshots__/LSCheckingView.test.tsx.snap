// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LSCheckingView/> Debe ser igual al snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-48qvd7-MuiGrid-root"
  >
    <div
      class="grid shadow-2xl shadow-black grid-cols-1 rounded-md bg-white p-4 space-x-4 animate-fade-up"
    >
      <div
        class="text-center"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1wj8ty9-MuiTypography-root"
        />
      </div>
      <div
        class="flex flex-col items-center"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-et5mm8-MuiTypography-root"
        >
          Checking your credentials, plase wait.
        </p>
        <span
          class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1wxecgq-MuiCircularProgress-root"
          role="progressbar"
          style="width: 50px; height: 50px;"
        >
          <svg
            class="MuiCircularProgress-svg css-54pwck-MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-19t5dcl-MuiCircularProgress-circle"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;
